### DISCUSSION TITLE
modifying the source code

### DISCUSSION THREAD

**Discussion by gekkehenkie11** [emporia-vue-local/esphome#285] [General]
So I want to do some experiments and send some custom raw data. I'm quite confused about how to modify the source code. Or where it's even located :) I'm running a home assistant partition via virtualbox, I assume it must be in there? But how do I find it and edit it?

**Comment by flaviut** [comment]
1. fork this repo
2. make your changes and push them to github
3. change `  - source: github://emporia-vue-local/esphome@dev` to point to your fork/branch
4. re-compile

**Comment by gekkehenkie11** [comment]
Hmm.. so I fork it, edit it, compile it. Works. However when I make changes for a second time and then make a clean build and then install, 
 it seems to not incorporate that last commit. It just keeps building the same code as I made it build the first time. Any ideas? And yes, I'm pointing yaml to my own fork :) In fact I can edit, compile, it works the first time. But it keeps rebuilding that first compilation. So it seems when I "install" it doesn't re-download everything from github.

**Reply by wadesherman** [reply]
use the "refresh" attribute in external_components: https://esphome.io/components/external_components.html
